src/granite/query/builder_methods.cr

```diff
@@ -6,7 +6,18 @@
 # end
 module Granite::Query::BuilderMethods
   def __builder
-    Builder(self).new
+    db_type = case adapter.class
+              when Granite::Adapter::Pg
+                DbType::Pg
+              when Granite::Adapter::Mysql
+                DbType::Mysql
+              when Granite::Adapter::Sqlite
+                DbType::Sqlite
+              else
+                raise "Adapter not supported #{Model.adapter.class}"
+              end
+
+    Builder(self).new(db_type)
   end
 
   delegate where, count, order, offset, limit, first, to: __builder
```

Comment:
@robacarp when I try to `extend Granite::Query::BuilderMethods`, I get an error for the `first` method since it's defined in the `querying.cr` module.  It looks like `count` is also defined in both.  

What is the plan for integration?  Did we want to move all the current `querying` functionality over?  Should we look at adding the lazy loaded `Collection` into the executor?